core/src/main/java/info/nightscout/androidaps/plugins/iob/iobCobCalculator/GlucoseStatusProvider.kt

```diff
@@ -100,8 +102,35 @@ class GlucoseStatusProvider @Inject constructor(
         ).also { aapsLogger.debug(LTag.GLUCOSE, it.log()) }.asRounded()
     }
 
-    companion object {
+    fun isBgFlatForInterval(minutes: Long, maxDelta: Double) : Boolean? {
```

Comment:
Missing the original isSaveCgmSource check here which is set to check if the source is Dexcom (BYODA)

Reply:
I didn't changed that check at all. As I've said it _can_ be changed to respect this smoothing option too, but imho that must be an additional change, subject for further discussion.

Reply:
Well in the old behavoir (in dev) the Dexcom plugin is excluded from the 45min unchanged check, this is not included in your PR as far as I can see. 

Current implementation in dev:

variable saveCgmSource is set here:
https://github.com/nightscout/AndroidAPS/blob/b4edb71e279954a1aa93a9ac720898d57494221e/app/src/main/java/info/nightscout/androidaps/plugins/aps/openAPSSMB/OpenAPSSMBPlugin.kt#L210

And is used in determine_basal.js here
https://github.com/nightscout/AndroidAPS/blob/b4edb71e279954a1aa93a9ac720898d57494221e/app/src/main/assets/OpenAPSSMB/determine-basal.js#L113

Reply:
ahh, this one. ok, i'll re-introduce it here too, ok

Reply:
pushed update

Reply:
the variable `isSaveCgmSource` is still used in determine-basal for some conditions so seems that it must be reviewed and replaced with something like `byoda || smoothed`, but there are still some points to discuss:

1. byoda with g7 does not have smoothing currently, so the check for byoda becomes not so trustful
2. this check must be moved somewhere else to be shared and not copy-pasted by plugins.